src/v/kafka/client/fetcher.cc

```diff
@@ -27,7 +28,7 @@ fetch_request make_fetch_request(
     std::vector<fetch_request::partition> partitions;
     partitions.push_back(fetch_request::partition{
       .partition_index{tp.partition},
-      .current_leader_epoch = 0,
+      .current_leader_epoch = kafka::invalid_leader_epoch,
```

Comment:
+1 here, this later needs to change to the value returned by the metadata response.

Reply:
I created an issue #2046 to track this in kafka client